pkg/api/runner.go

```diff
@@ -26,6 +35,11 @@ func pool(fn Recorder, num int) Recorder {
 
 		for i := 0; i < num; i++ {
 			go func() {
+				defer func() {
+					if r := recover(); r != nil {
+						errCh <- PanicError{p: fmt.Sprintf("panic: %s", r)}
+					}
```

Comment:
nit: this if statement is duplicated 4 times in this PR, is there a way to refactor it and reuse it?